azuredevops/internal/service/serviceendpoint/resource_serviceendpoint_nexus.go

```diff
@@ -0,0 +1,115 @@
+package serviceendpoint
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/google/uuid"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+	"github.com/microsoft/azure-devops-go-api/azuredevops/v6/serviceendpoint"
+	"github.com/microsoft/terraform-provider-azuredevops/azuredevops/internal/utils/converter"
+	"github.com/microsoft/terraform-provider-azuredevops/azuredevops/internal/utils/tfhelper"
+)
+
+// ResourceServiceEndpointNexus schema and implementation for Nexus service endpoint resource
+func ResourceServiceEndpointNexus() *schema.Resource {
+	r := genBaseServiceEndpointResource(flattenServiceEndpointNexus, expandServiceEndpointNexus)
+
+	r.Schema["url"] = &schema.Schema{
+		Type:     schema.TypeString,
+		Required: true,
+		ValidateFunc: func(i interface{}, key string) (_ []string, errors []error) {
+			url, ok := i.(string)
+			if !ok {
+				errors = append(errors, fmt.Errorf("expected type of %q to be string", key))
+				return
+			}
+			if strings.HasSuffix(url, "/") {
+				errors = append(errors, fmt.Errorf("%q should not end with slash, got %q.", key, url))
+				return
+			}
+			return validation.IsURLWithHTTPorHTTPS(url, key)
+		},
+		Description: "Url for the Nexus Repository",
+	}
+
+	patHashKeyP, patHashSchemaP := tfhelper.GenerateSecreteMemoSchema("password")
+	aup := &schema.Resource{
+		Schema: map[string]*schema.Schema{
+			"username": {
+				Description: "The Nexus user name.",
+				Type:        schema.TypeString,
+				Required:    true,
+				Sensitive:   true,
+			},
+			"password": {
+				Description:      "The Nexus password.",
+				Type:             schema.TypeString,
+				Required:         true,
+				Sensitive:        true,
+				DiffSuppressFunc: tfhelper.DiffFuncSuppressSecretChanged,
+			},
+			patHashKeyP: patHashSchemaP,
+		},
+	}
+
+	r.Schema["authentication_basic"] = &schema.Schema{
+		Type:     schema.TypeList,
+		Required: true,
+		MinItems: 1,
+		MaxItems: 1,
+		Elem:     aup,
+	}
```

Comment:
1. Hash functions can be removed. These functions will be deprecated in the future.
2. `usname/pwd` is the only auth type supported by this resource, can we flatten `username` and `password` to the the root, not wrap by `authentication_basic`
	
```suggestion
	r.Schema["username"] = &schema.Schema{
		Description: "The Nexus user name.",
		Type:        schema.TypeString,
		Required:    true,
		Sensitive:   true,
	}

	r.Schema["password"] = &schema.Schema{
		Description: "The Nexus password.",
		Type:        schema.TypeString,
		Required:    true,
		Sensitive:   true,
	}
```

Reply:
Done